README.md

```diff
@@ -4,12 +4,57 @@ XenAPI Software Development Kit
 This repository contains the generation code for the XenAPI Software Development
 Kit (SDK) and its associated documentation and examples.
 
-The generated SDK consists of five components, one for each of C, C#, Java,
-PowerShell and Python, exposing the individual XenAPI calls as first-class
-functions in the target language.
+The SDK consists of five components, one for each of C, C#, Java, PowerShell,
+and Python, exposing the individual XenAPI calls as first-class functions in the
+target language.
 
-The generation code is mostly written in
-[Ocaml](http://caml.inria.fr/ocaml/) 3.12.
+The generation code is mostly written in [Ocaml][1] 3.12.
+
+Build
+-----
+
+#### 1. Environment setup
+
+Install [opam][2]. It is also recommended to install one of the listed external
+solvers (e.g. aspcud).
+
+Ensure you are using version 4.02.3 of the Ocaml compiler:
+`opam switch 4.02.3`
+
+and you have the package depext: `opam install depext`
+
+_Note: if you want to avoid setting up the environment by yourself, you can use
+one of the ready containers from [opam-dockerfiles][3], which come with a
+preinstalled OPAM environment and Ocaml compiler._
+
+#### 2. Install dependencies
+
+Add [xs-opam][4] as a remote Opam repository (follow the instructions in that
+repo).
+
+The SDK is dependent on the XenAPI, therefore run
+
+```
+   opam depext -y xapi
+   opam install -y xapi
+```
+This will install first the dependencies of the xapi package and then the package
+itself following its master version.
+
+Finally, copy [this file][5] into the root of the xen-api-sdk repo.
+
+#### 3. SDK generation
+
+Run `make <lang>`, where `<lang>` one or more of `c`, `java`, `csharp`,
+`powershell`, to generate SDK source code for the specified programming
+language(s), or simply `make` to generate SDK source code for all the supported
+languages.
+
+To compile the generated source code, follow the instructions in the corresponding
+README files.
+
+_Note: The Python module is not auto-generated. It can be found at
```

Comment:
I used the exact location because people might not think of looking into the examples folder 😉